Added option and method to check for view setting

andrasguseo opened the pull request:
I needed this for one of the extension and thought it might be useful in the template.
Feel free to adjust it or totally scrap it.
Also I'm happy to receive feedback.
Note: if we want to roll this in the template fast, then probably it's best if it's not me doing further adjustments. :)

Commit:
Added option and method to check for view setting

I needed this for one of the extension and thought it might be useful in the template.
Feel free to adjust it or totally scrap it.

Commit:
Housekeeping :)

mitogh commented on tribe-ext-extension-template.php:
```diff
@@ -204,6 +222,47 @@ private function php_version_check() {
+			$view_breaks_version = $view_required_version == 'V1 legacy' ? 'V2 updated' : 'V1 legacy';
+
+			$show_warning = (
+				$view_required_version == 'V1 legacy' && tribe_events_views_v2_is_enabled()
```
```suggestion
				$view_required_version === 'V1 legacy' && tribe_events_views_v2_is_enabled()
```

mitogh commented on tribe-ext-extension-template.php:
```diff
@@ -204,6 +222,47 @@ private function php_version_check() {
+
+			$show_warning = (
+				$view_required_version == 'V1 legacy' && tribe_events_views_v2_is_enabled()
+				|| $view_required_version == 'V2 updated' && ! tribe_events_views_v2_is_enabled()
```
```suggestion
				|| $view_required_version === 'V2 updated' && ! tribe_events_views_v2_is_enabled()
```

mitogh commented on tribe-ext-extension-template.php:
```diff
@@ -56,6 +56,20 @@ class Main extends Tribe__Extension {
+		 *
+		 * @todo Remove this and corresponding method v2_views_check() if extension works with both V1 and V2 views.
+		 */
+		public $view_needed = "V1 legacy";
```
Maybe 

```suggestion
		public $required_view = "V1 legacy";
```

mitogh commented on tribe-ext-extension-template.php:
```diff
@@ -204,6 +222,47 @@ private function php_version_check() {
+				}
+
+				return false;
+			}
```
To avoid large indentation.

```suggestion
			if ( ! function_exists( 'tribe_events_views_v2_is_enabled' ) ) {
				return true;
			}
			
			if ( is_admin() && current_user_can( 'activate_plugins' ) && $show_warning ) {
					$message = '<p>';
					$message .= sprintf(
						__(
							'%s requires %s views to work. It will not work with %s views.', PLUGIN_TEXT_DOMAIN
						),
						$this->get_name(),
						$view_required_version,
						$view_breaks_version,
					);
					$message .= '</p>';

					tribe_notice( PLUGIN_TEXT_DOMAIN . '-views-version', $message, [ 'type' => 'warning' ] );
				}
			return false;
```

mitogh commented on tribe-ext-extension-template.php:
```diff
@@ -204,6 +222,47 @@ private function php_version_check() {
+					);
+					$message .= '</p>';
+
+					tribe_notice( PLUGIN_TEXT_DOMAIN . '-views-version', $message, [ 'type' => 'warning' ] );
```
I think `PLUGIN_TEXT_DOMAIN` is no longer available over here.

andrasguseo replied:
Right. Fixed it.

Commit:
Update tribe-ext-extension-template.php

Co-Authored-By: Crisoforo Gaspar Hernández <hello@crisoforo.com>

Commit:
Update tribe-ext-extension-template.php

Co-Authored-By: Crisoforo Gaspar Hernández <hello@crisoforo.com>

Commit:
Update tribe-ext-extension-template.php

Co-Authored-By: Crisoforo Gaspar Hernández <hello@crisoforo.com>

Commit:
Update tribe-ext-extension-template.php

Ah, makes sense.

Co-Authored-By: Crisoforo Gaspar Hernández <hello@crisoforo.com>

Commit:
corrected text domain

Commit:
adjusted parameter name

andrasguseo commented:
@mitogh Thanks for all your great feedback!
I adjusted whatever was needed.
Please re-check this when you finish your plate,
So we can merge it into the template.
:)

cliffordp commented on tribe-ext-extension-template.php:
```diff
@@ -56,6 +56,20 @@ class Main extends Tribe__Extension {
+		 *
+		 * @todo Remove this and corresponding method v2_views_check() if extension works with both V1 and V2 views.
+		 */
+		public $required_view = "V1 legacy";
```
single quotes, not double

cliffordp replied:
why so lengthy? why not just `1` / `2` or `v1` / `v2` (`v` or `V`)?

cliffordp replied:
or maybe make it an array of views it _does_ work with?

andrasguseo replied:
I used a full string because that string is then used in the warning message. I'll think about this some more.
[...]
Using array / listing all compatible views is not really needed. If it works with both views, then we can just delete that part of the template (like we do with other parts).
Apart from that it's either one version or the other.

cliffordp commented on tribe-ext-extension-template.php:
```diff
@@ -204,6 +222,48 @@ private function php_version_check() {
+
+			$view_breaks_version = $view_required_version == 'V1 legacy' ? 'V2 updated' : 'V1 legacy';
+
+			$show_warning = (
+				$view_required_version === 'V1 legacy' && tribe_events_views_v2_is_enabled()
+				|| $view_required_version === 'V2 updated' && ! tribe_events_views_v2_is_enabled()
+			)
+				? true
+				: false;
```
avoid ternaries, especially multi-line, to be more legible -- and each condition on its own line

cliffordp commented on tribe-ext-extension-template.php:
```diff
@@ -204,6 +222,48 @@ private function php_version_check() {
+				return true;
+			}
+			
+			if ( is_admin() && current_user_can( 'activate_plugins' ) && $show_warning ) {
```
each condition on its own line

cliffordp commented on tribe-ext-extension-template.php:
```diff
@@ -204,6 +222,48 @@ private function php_version_check() {
+					$message = '<p>';
+					$message .= sprintf(
+						__(
+							'%s requires %s views to work. It will not work with %s views.', PLUGIN_TEXT_DOMAIN
```
fix `PLUGIN_TEXT_DOMAIN`

cliffordp commented on tribe-ext-extension-template.php:
```diff
@@ -204,6 +222,48 @@ private function php_version_check() {
+					);
+					$message .= '</p>';
+
+					tribe_notice( 'tribe-ext-extension-template' . '-views-version', $message, [ 'type' => 'warning' ] );
```
the `'tribe-ext-extension-template' . '-views-version'` format is leftover from replacing `PLUGIN_TEXT_DOMAIN` -- please fix to be all one single string, here and in the template's other places it's like this

cliffordp reviewed (dismissed):
good job. looking forward to re-reviewing after these changes.

Commit:
Splitting up the ternary.

Wasn't totally sure about the formating.

Commit:
Each condition on its own line

And removed duplicate return.

Commit:
Fixing text domains

Commit:
Made the version check a bit simpler

I am using 'V1' and 'V2' to keep the clarity, as using only '1' and '2' might get confusing down the road.
The needed strings for the tribe notice are being set up in the method.

Commit:
Added a notice for developers if `$required_view` is not set up properly.

Commit:
Made strings translatable

cliffordp committed:
remove dup prop

cliffordp committed:
refactored v1/v2 check and notice

cliffordp commented:
I reworked it.

Here's a 3min video of the notice working: https://share.getcloudapp.com/lluDQ8b1

And a screenshot of one of them: 
![image](https://user-images.githubusercontent.com/1812179/79250459-42188300-7e44-11ea-8c86-627d5a0ee39b.png)
